src/UnoFlipController.java

```diff
@@ -37,93 +33,125 @@ public UnoFlipController(UnoFlipModel model){
+                    int numPlayers = 0;
+
                     String option = (String) comboBox.getSelectedItem();
-                    colour = Card.Colour.valueOf(option);
-                    this.model.setCurrentColour(colour);
-                    this.model.notifyViews();
-                    break;
-                } else if (result == JOptionPane.CLOSED_OPTION){
-                    continue;
+                    numPlayers = Integer.parseInt(option);
+
+                    this.model.setNumPlayers(numPlayers);
+                } else { // Exit the game if the user cancels, absolutely must select an option, otherwise, do not start the game
+                    System.exit(0);
                 }
-            }
 
-        } else if(e.getActionCommand().equals(UnoFlipViewFrame.DRAW_CMD)){
-            this.model.playTurn(-1);
-        }
-        else if(e.getActionCommand().equals(UnoFlipViewFrame.NEXT_CMD)) {
-            this.model.nextTurn();
-        } else if (e.getActionCommand().equals(UnoFlipViewFrame.CHALLENGE_CMD) ) {
+                // Prompt the user for the name of each player
+                for (int i = 0; i < this.model.getNumPlayers(); i++) {
+                    String name = (String) JOptionPane.showInputDialog("Enter player " + Integer.toString(i+1) + " name");
+
+                    // If player does not select name, give a default name (Player + player number) ex. Player 2
+                    if (name == null || name.equals("")) {
+                        this.model.createPlayer("Player " + (i+1));
+                    }
+                    else {
+                        this.model.createPlayer(name);
+                    }
+                }
 
-            boolean challenge;
-            int result = JOptionPane.showConfirmDialog(null, "Do you want to continue?", "Confirmation", JOptionPane.YES_NO_OPTION);
+                this.model.setUpInitialTopCard();
+                break;
+
+            // User places a wild card and prompts the player to choose a colour to update game state
+            case UnoFlipViewFrame.WILD_CMD:
+                // Colour options for Player to chose from
+                String[] colourOptions = {"RED", "BLUE", "YELLOW", "GREEN"};
+
+                comboBox = new JComboBox<>(colourOptions);
+                // Keep asking user to pick a colour until they do
+                while(true) {
+                    result = JOptionPane.showOptionDialog(
+                            null,
+                            comboBox,
+                            "Select colour:",
+                            JOptionPane.DEFAULT_OPTION,
+                            JOptionPane.PLAIN_MESSAGE,
+                            null,
+                            null,
+                            null);
+
+                    // Colour is selected and OK is pressed
+                    if (result == JOptionPane.OK_OPTION) {
+                        // Colour to set the new current colour of the game
```

Comment:
there are still redundant comments in this block, you don't need to say all that